routes/emailAuthData.js

```diff
@@ -0,0 +1,71 @@
+const log = require("loglevel");
+const express = require("express");
+
+const pify = require("pify");
+const { getError } = require("../utils");
+const { validationMiddleware } = require("../middleware");
+const { redisClient } = require("../database");
+
+const router = express.Router();
+const redis = pify(redisClient);
+const REDIS_TIMEOUT = 300; // 5m
+const REDIS_NAME_SPACE = "EMAIL_AUTH_DATA";
+module.exports = (io) => {
+  io.on("connection", (socket) => {
+    socket.on("check_auth_status", async (channelId) => {
+      if (channelId) {
+        const instancePubKey = channelId;
+        const key = `${REDIS_NAME_SPACE}_${instancePubKey}`;
+        // check if data for pubKey already in db,
+        // if data exists then emit data and so that client
+        // will close the connection.
+        const data = await redis.get(key);
+        if (data) {
+          socket.emit("success", JSON.parse(data.value.encAuthData));
+        } else {
+          socket.join(instancePubKey);
+        }
+        // create a socket room , specific to instancePubKey
+      } else {
+        socket.emit("error", "Empty channel id id not allowed");
+      }
+    });
+  });
+
+  router.get("/fetchAuthData", validationMiddleware(["instancePubKey"], false), async (req, res) => {
+    try {
+      const { instancePubKey } = req.query;
+      const key = `${REDIS_NAME_SPACE}_${instancePubKey}`;
+      const data = await redis.get(key);
+      const value = (data && JSON.parse(data)) || "";
```

Comment:
Shouldn't this fail if data doesn't exist/ un-parsable ?

Reply:
if data doesn't exist it is returning empty string which is a falsy value and assumption is that data must always be parseable since data is inserted from only one place as stringified json  that is updateAuthData api and we are not using this api so may be we can remove it. Since we are fetching data only via sockets and updating it via api. @rathishubham7

Reply:
Please remove the API if not needed.